providers/aws/services/s3/s3_service.py

```diff
@@ -0,0 +1,100 @@
+import threading
+from dataclasses import dataclass
+
+from lib.logger import logger
+from providers.aws.aws_provider import current_audit_info, generate_regional_clients
+
+
+################## S3
+class S3:
+    def __init__(self, audit_info):
+        self.service = "s3"
+        self.session = audit_info.audit_session
+        self.audited_account = audit_info.audited_account
+        self.regional_clients = generate_regional_clients(self.service, audit_info)
+        self.__threading_call__(self.__list_buckets__)
+        self.__threading_call__(self.__get_bucket_versioning__)
+        self.__threading_call__(self.__get_bucket_logging__)
+
+    def __get_session__(self):
+        return self.session
+
+    def __threading_call__(self, call):
+        threads = []
+        for regional_client in self.regional_clients:
+            threads.append(threading.Thread(target=call, args=(regional_client,)))
+        for t in threads:
+            t.start()
+        for t in threads:
+            t.join()
+
+    def __list_buckets__(self, regional_client):
+        logger.info("S3 - Listing buckets...")
+        try:
+            list_buckets = regional_client.list_buckets()
```

Comment:
Why if you query AWS only once to retrieve the bucket list?

Reply:
At the moment, the only solution to speed up this algorithm is to have regional resources, which is going to be done in the future.